pkg/controllers/federatedhpa/federatedhpa_controller.go

```diff
@@ -678,6 +679,8 @@ func (c *FederatedHPAController) validateAndParseSelector(hpa *autoscalingv1alph
 
 // Computes the desired number of replicas for a specific hpa and metric specification,
 // returning the metric status and a proposed condition to be set on the HPA object.
+//
+//nolint:gocyclo
 func (c *FederatedHPAController) computeReplicasForMetric(ctx context.Context, hpa *autoscalingv1alpha1.FederatedHPA, spec autoscalingv2.MetricSpec,
```

Comment:
Too many parameters on this method, please open an issue to optimize it and remove `nolint:gocyclo`.